src/main/scala/org/constellation/domain/consensus/ConsensusService.scala

```diff
@@ -144,6 +144,13 @@ abstract class ConsensusService[F[_]: Concurrent, A <: ConsensusObject] extends
         )
       )
 
+  def findByPredicate(predicate: A => Boolean): F[List[A]] =
+    for {
+      p <- pending.toMap()
+      a <- accepted.toMap()
+      ic <- inConsensus.toMap()
+    } yield (p ++ a ++ ic).values.filter(predicate).toList
+
```

Comment:
Don't you think that it will be better to firstly filter out by predicate and then merge together?

Reply:
I thought about it and I decided against it. It would either result in duplication of the code or in an increased complexity. Complexity would increase because of lifted functions, we would have to reduce List[F[Map]]  to F[List] to aviod duplication. What do you think, should I still change it?

Reply:
Hm, taking into consideration the amount of data and merging thousands of data together or merging just a few, then the latter is probably a better choice, even though some duplication of code.

```
val filter: Map[String, A] => List[A] = _.filter(predicate).values.toList

for {
 p <- pending.toMap().map(filter)
 a <- accepted.toMap().map(filter)
 ic <- inConsensus.toMap().map(filter)
} yield p ++ a ++ ic
```

@kpudlik @TheMMaciek any ideas?

Reply:
Your code looks nice, I will apply these changes.